include/moqt/core/transport.h

```diff
@@ -0,0 +1,226 @@
+                                             std::shared_ptr<PublishTrackHandler> track_delegate);
+
+        /**
+         * @brief Unpublish track
+         *
+         * @param conn_id           Connection ID from transport for the QUIC connection context
+         * @param track_delegate    Track delegate used when published track
+         */
+        void UnpublishTrack(TransportConnId conn_id, std::shared_ptr<PublishTrackHandler> track_delegate);
+
+        /**
+         * @brief Get the instance status
+         *
+         * @return Status indicating the state/status of the instance
+         */
+        Status GetStatus();
+
+        // --------------------------------------------------------------------------
+        // Metrics
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Connection metrics for server accepted connections
+         *
+         * @details Connection metrics are updated real-time and transport quic metrics on
+         *      Config::metrics_sample_ms period
+         */
+        std::map<TransportConnId, ConnectionMetrics> connection_metrics_;
+
+      private:
+        // -------------------------------------------------------------------------------------------------
+        // Transport Delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        void on_new_data_context([[maybe_unused]] const TransportConnId& conn_id,
+                                 [[maybe_unused]] const DataContextId& data_ctx_id) override
+        {
+        }
+
+        void on_connection_status(const TransportConnId& conn_id, const TransportStatus status) override;
+        void on_new_connection(const TransportConnId& conn_id, const TransportRemote& remote) override;
+        void on_recv_stream(const TransportConnId& conn_id,
+                            uint64_t stream_id,
+                            std::optional<DataContextId> data_ctx_id,
+                            const bool is_bidir = false) override;
+        void on_recv_dgram(const TransportConnId& conn_id, std::optional<DataContextId> data_ctx_id) override;
+
+        // -------------------------------------------------------------------------------------------------
+        // End of transport delegate/callback functions
+        // -------------------------------------------------------------------------------------------------
+
+        struct ConnectionContext
+        {
+            TransportConnId conn_id;
+            std::optional<uint64_t> ctrl_data_ctx_id;
+            bool setup_complete{ false }; ///< True if both client and server setup messages have completed
+            uint64_t client_version{ 0 };
+            std::optional<messages::MoqtMessageType>
+              ctrl_msg_type_received; ///< Indicates the current message type being read
```

Comment:
can it be data message too ? If so we should rename the variable name

Reply:
Private members, such as the above are not part of the public API.  These are internal use member variables that are for the implementation.